### PR TITLE
Support windows targets

### PR DESCRIPTION
**Pull request by emesare**
Tested on windows 10 and windows 11.

EDIT: It seems there is something else within the extraction that relies on windows symlinks directly which is an issue for non-admin users:

```
15:01:14  Caused by:
15:01:14    process didn't exit successfully: `C:/target\release\build\flatbuffers-build-a81ddc4b08ecd034\build-script-build` (exit code: 101)
15:01:14    --- stderr
15:01:14    thread 'main' panicked at C:\.cargo\git\checkouts\flatbuffers-build-8ef4ff77bea6d0f6\46e481f\build.rs:5:30:
15:01:14    failed to vendor flatc: failed to unpack `\\?\C:\tmp\.tmp6bPG6a\flatbuffers\flatbuffers-24.3.25\docs\source\CONTRIBUTING.md`
15:01:14  
15:01:15    Caused by:
15:01:15        A required privilege is not held by the client. (os error 1314) when symlinking ../../CONTRIBUTING.md to \\?\C:\tmp\.tmp6bPG6a\flatbuffers\flatbuffers-24.3.25\docs\source\CONTRIBUTING.md
```

### CODE REVIEW

**Comment by carlocorradini** [comment]
@rdelfin Any update?
I need this 😢

**Comment by rdelfin** [comment]
oh! Really sorry, didn't see your PR! I'll take a look and prepare a release

File: src/lib.rs [hunk 1]
```diff
@@ -310,7 +310,10 @@ fn generate_symlink<P: AsRef<Path>, Q: AsRef<Path>>(symlink_path: P, output_path
     if symlink_path.as_ref().exists() {
         std::fs::remove_file(&symlink_path).map_err(Error::SymlinkCreationFailure)?;
     }
+    #[cfg(unix)]
     std::os::unix::fs::symlink(output_path, symlink_path).map_err(Error::SymlinkCreationFailure)?;
+    #[cfg(windows)]
+    junction::create(output_path, symlink_path).map_err(Error::SymlinkCreationFailure)?;
```
Comment (rdelfin) [inline src/lib.rs:316]: Looks like the they're trying to add support on the standard library, but it's still nightly :/
https://github.com/rust-lang/rust/issues/121709

**Review by rdelfin — APPROVED** [review]
Thanks for taking the time to do this!

**Comment by rdelfin** [comment]
Fixing clippy error in #12

**Comment by carlocorradini** [comment]
@rdelfin It works! Thanks! 🥳

**Comment by rdelfin** [comment]
Glad to @carlocorradini. Check [this project's crates.io](https://crates.io/crates/flatbuffers-build) and you should see a new version (0.2.1) with windows support

File: src/lib.rs (hunk 1 above)
Comment (emesare) [reply src/lib.rs:317]: Yea I have a requirement to run on rust 1.77 so I couldn't do that

Comment (rdelfin) [reply src/lib.rs:317]: Yeah absolutely, I don't want to turn this into a nightly-only crate

**Pull request merged**